### PR TITLE
Response as GeoJSON

### PR DESCRIPTION
**Pull request by manueltimita**
As discussed in https://github.com/graphhopper/graphhopper/issues/1591, I have added the possibility to make a request to the isochrone endpoint specifying the response type as GeoJSON with `type=geojson`. Using `type=json` or no `type` parameter returns the output as usual.

### CODE REVIEW

**Comment by karussell** [comment]
Can you update this to the latest master? (there seems to be some conflicts)

**Comment by manueltimita** [comment]
> Can you update this to the latest master? (there seems to be some conflicts)

Fixed

File: web/src/test/java/com/graphhopper/http/isochrone/IsochroneResourceTest.java [hunk 1]
```diff
@@ -117,7 +120,87 @@ public void requestWithShortest() throws Exception {
...
+        assertEquals(lastFeature.path("geometry").path("type").asText(), "Polygon");
+    }    
+    
+    private Response requestIsochrone(String path) {
+    	String url = "http://localhost:8080" + path;
+        return app.client().target(url).request().buildGet().invoke();
+    }
+    
+    private JsonNode parseRequestResponse (Response response) throws IOException {
+    	response.bufferEntity();
```
Comment (karussell) [inline web/src/test/java/com/graphhopper/http/isochrone/IsochroneResourceTest.java]: Why is this necessary?

File: web-bundle/src/main/java/com/graphhopper/resources/IsochroneResource.java [hunk 2]
```diff
@@ -117,15 +122,31 @@ public Response doGet(
                 JsonFeature feature = new JsonFeature();
                 HashMap<String, Object> properties = new HashMap<>();
                 properties.put("bucket", features.size());
+                if (respType.equalsIgnoreCase("geojson")) {
+                    // If you replace GraphHopper with your own brand name, this is fine.
+                    // Still it would be highly appreciated if you mention us in your about page!
+                    String[] copyrights = {"GraphHopper", "OpenStreetMap contributors"};
+                    properties.put("copyrights", copyrights);
```
Comment (karussell) [inline web-bundle/src/main/java/com/graphhopper/resources/IsochroneResource.java]: I have slightly updated master to reduce this to one line:

properties.put("copyrights", WebHelper.COPYRIGHTS);

Can you update to master and also remove the comment?

File: web/src/test/java/com/graphhopper/http/isochrone/IsochroneResourceTest.java (hunk 1 above)
Comment (manueltimita) [reply web/src/test/java/com/graphhopper/http/isochrone/IsochroneResourceTest.java]: My bad, this is a leftover from when I was calling `readEntity` twice. I will fix and resubmit

File: web-bundle/src/main/java/com/graphhopper/resources/IsochroneResource.java (hunk 2 above)
Comment (manueltimita) [reply web-bundle/src/main/java/com/graphhopper/resources/IsochroneResource.java]: Done

**Comment by karussell** [comment]
Nice, thanks!

@manueltimita did you test this for leaflet or some different client?

@nilsnolde would you have time to try this for your use case? Before providing a new format in the public JSON API I would like to make sure it works for at least two different use cases (i.e. clients). I mean GeoJSON is standardized and everything but just to make sure it works...

**Comment by manueltimita** [comment]
> @manueltimita did you test this for leaflet or some different client?

I don't have a **graphopper** installation, but I did check the GeoJSON tests output against the standard and saw no problems. Here's an example of how to check for yourself if you need to: 

Test 1: Run https://github.com/manueltimita/graphhopper/blob/response_as_geojson/web/src/test/java/com/graphhopper/http/isochrone/IsochroneResourceTest.java#L152-L172 and copy/paste the console output (`System.out.println(json);`) into http://geojson.io or http://geojsonlint.com/. They are both using https://github.com/mapbox/mapbox.js/ but if it works there it will definitely work with [Leaflet GeoJSON](https://leafletjs.com/examples/geojson/). Here's the output using Leaflet: https://jsfiddle.net/e2zuqhtp/ 

Test 2 (multiple buckets):
https://github.com/manueltimita/graphhopper/blob/response_as_geojson/web/src/test/java/com/graphhopper/http/isochrone/IsochroneResourceTest.java#L175-L189 and here's the fiddle with the output in Leaflet 1.0.3: https://jsfiddle.net/t1bsLqu8/1/ (in this one, you can click on the buckets)

**Comment by karussell** [comment]
Thanks! I think we should test it with openlayers as a second way, just to make sure. Feel free to do this or I'll try later (not sure yet when though)

**Comment by manueltimita** [comment]
Here is how `requestIsochrone("/isochrone?point=42.531073,1.573792&time_limit=130&type=geojson&buckets=3")` looks like on OL v4: https://jsfiddle.net/6gp1v2cn/

**Comment by nilsnolde** [comment]
Oops, sorry @karussell I just saw this.. Great stuff @manueltimita! Seems like you tested already, output looks good.

I'll try to find time to try with `routing-py`.

**Comment by karussell** [comment]
Thanks a lot @manueltimita - this looks good! Will merge.

@nilsnolde would be cool, if you can test this before 0.13 release (in ~6 weeks).

**Comment by nilsnolde** [comment]
Tested with [routingpy](https://github.com/gis-ops/routing-py), all good with `folium` maps, eats the GeoJSON without any parsing :+1: 

Will release soon on our side.
